src/binding/python/Series.cpp

```diff
@@ -20,17 +20,93 @@
  */
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
+#if openPMD_HAVE_MPI
+//  re-implemented signatures:
+//  include <mpi4py/mpi4py.h>
+#   include <mpi.h>
+#endif
 
 #include "openPMD/Series.hpp"
+#include <string>
 
 namespace py = pybind11;
 using namespace openPMD;
 
+#if openPMD_HAVE_MPI
+    /** mpi4py communicator wrapper
+     *
+     * refs:
+     * - https://github.com/mpi4py/mpi4py/blob/3.0.0/src/mpi4py/libmpi.pxd#L35-L36
+     * - https://github.com/mpi4py/mpi4py/blob/3.0.0/src/mpi4py/MPI.pxd#L100-L105
+     * - installed: include/mpi4py/mpi4py.MPI.h
+     */
+    struct openPMD_PyMPICommObject
+    {
+        PyObject_HEAD
+        MPI_Comm ob_mpi;
+        unsigned int flags;
+    };
+    using openPMD_PyMPIIntracommObject = openPMD_PyMPICommObject;
+#endif
+
 
 void init_Series(py::module &m) {
     py::class_<Series, Attributable>(m, "Series")
 
-        .def(py::init<std::string const&, AccessType>())
+        .def(py::init<std::string const&, AccessType>(),
+            py::arg("filepath"), py::arg("access_type"))
+#if openPMD_HAVE_MPI
+        .def(py::init([](std::string const& filepath, AccessType at, py::object &comm){
+            //! @todo perform mpi4py import test and check min-version
+            //!       careful: double MPI_Init risk? only import mpi4py.MPI?
+            //!       required C-API init? probably just checks:
+            //! refs:
+            //! - https://bitbucket.org/mpi4py/mpi4py/src/3.0.0/demo/wrap-c/helloworld.c
+            //! - installed: include/mpi4py/mpi4py.MPI_api.h
+            // if( import_mpi4py() < 0 ) { here be dragons }
+
+            if( comm.ptr() == Py_None )
+                throw std::runtime_error("Series: MPI communicator cannot be None.");
+            if( comm.ptr() == nullptr )
+                throw std::runtime_error("Series: MPI communicator is a nullptr.");
+
+            // check type string to see if this is mpi4py
+            //   __str__ (pretty)
+            //   __repr__ (unambiguous)
+            //   mpi4py: <mpi4py.MPI.Intracomm object at 0x7f998e6e28d0>
+            //   pyMPI:  ... (todo)
+            py::str const comm_pystr = py::repr(comm);
+            std::string const comm_str = comm_pystr.cast<std::string>();
+            if( comm_str.substr(0, 12) != std::string("<mpi4py.MPI.") )
```

Comment:
Should probably only match `mpi4py` Communicators ([and derived](https://github.com/mpi4py/mpi4py/blob/3.0.0/src/mpi4py/MPI.pxd#L100-L141)/same-layouted objects)